detector/json_detection_results.go

```diff
@@ -0,0 +1,72 @@
+package detector
+
+import "talisman/git_repo"
+
+type FailureDetails struct {
```

Comment:
FailureDetails struct is also used as warning details. Could we rename this to something else ?

Reply:
Have implemented the changes as requested earlier